kvbc/include/direct_kv_db_adapter.h

```diff
@@ -54,21 +54,9 @@ class DBKeyManipulator : public DBKeyManipulatorBase {
   static concordUtils::Sliver extractKeyFromMetadataKey(const Key &_composedKey);
   static bool isKeyContainBlockId(const Key &_composedKey);
   static KeyValuePair composedToSimple(KeyValuePair _p);
-  static concordUtils::Sliver generateMetadataKey(storage::ObjectId objectId);
-  static concordUtils::Sliver generateStateTransferKey(storage::ObjectId objectId);
-  static concordUtils::Sliver generateSTPendingPageKey(uint32_t pageid);
-  static concordUtils::Sliver generateSTCheckpointDescriptorKey(uint64_t chkpt);
-  static concordUtils::Sliver generateSTReservedPageStaticKey(uint32_t pageid, uint64_t chkpt);
-  static concordUtils::Sliver generateSTReservedPageDynamicKey(uint32_t pageid, uint64_t chkpt);
-  static uint64_t extractCheckPointFromKey(const char *_key_data, size_t _key_length);
-  static std::pair<uint32_t, uint64_t> extractPageIdAndCheckpointFromKey(const char *_key_data, size_t _key_length);
-
- private:
-  static concordUtils::Sliver generateReservedPageKey(EDBKeyType, uint32_t pageid, uint64_t chkpt);
-  static bool copyToAndAdvance(char *_buf, size_t *_offset, size_t _maxOffset, char *_src, size_t _srcSize);
 };
 
-class DBAdapter : public DBAdapterBase {
+class DBAdapter : public IDbAdapter, public DBAdapterBase {
```

Comment:
Can we avoid the use of multiple inheritance? It's can be especially confusing to inherit from an abstract class and a concrete class. Maybe in a future commit we can pull out all the shared information in DBAdapterBase and instead use composition.

Reply:
Sure, we have already discussed this with @dartdart26